site-src/geps/gep-957.md

```diff
@@ -0,0 +1,140 @@
+# GEP-957: Destination Port Matching
+
+* Issue: [#957](https://github.com/kubernetes-sigs/gateway-api/issues/957)
+* Status: Provisional
+
+## TLDR
+
+Add a new `port` field to ParentRef to support port matching in Routes.
+
+## Goals
+
+* Support port matching in routes based on the destination port number of the
+  request.
+
+## Non-Goals
+
+* Support port matching based on port name.
+
+## Introduction
+
+Port matching is a common service mesh use case where traffic policies/rules
+need to be applied to traffic to certain destination ports. For ingress, while
+the API today already supports attaching a route to a specific listener, it may
+be useful to support attaching routes to listener(s) on a specified port. This
+allows route authors to apply networking behaviors on a fixed port.
+
+## API
+
+The proposal is to add a new field `Port` to `ParentRef`:
+
+```go
+type ParentRef struct {
+  ...
+  // Port is the network port this resource targets. It can be interpreted
+  // differently based on the type of parent resource:
+  //
+  // Gateway: All listeners listening on the specified port that also support
+  // this kind of Route. It's only recommended to set `Port` when the networking
+  // behaviors specified in a Route should only apply to a specific port as
+  // opposed to listener(s) whose port(s) may get switched by Gateway admins.
+  // When both Port and SectionName are specified, the name and port of the
+  // selected listener must match both specified values.
+  //
+  // Implementations MAY choose to support other parent resources.
+  // If that is the case, they MUST clearly document how Port is
+  // interpreted.
+  //
+  // When both Port and SectionName are unspecified, this will reference the
+  // entire resource.
```

Comment:
Recommend leaving this out. Port field is a more specific matcher and should refrain from general behavior guidance.